en/components/circular_progress.md

```diff
@@ -41,202 +33,240 @@ import { IgxProgressBarModule } from 'igniteui-angular';
 </div>
 <div>
 <button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="indeterminate-progressbar-sample-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
 </div>
 <div class="divider--half"></div>
 
-### Styling
+#### Dynamic Progress
 
-To get started with styling the circular progress bar, we need to import the `index` file, where all the theme functions and component mixins live:
+You can dynamically change the progress value of the circular progress bar, using controls to change the value of the progress. We can achieve that by binding the value with a custom variable:
 
-```scss
-@import '~igniteui-angular/lib/core/styles/themes/index';
-``` 
+```html
+<div class="sample-content">
+    <igx-circular-bar
+    [value]="currentValue"
+    [max]="100"
+    [animate]="true"
+    class="custom-size"
+    >
+    <div class="button-container">
+        <button igxButton="icon" (click)="removeProgress()">
+            <igx-icon fontSet="material">remove</igx-icon>
+        </button>
+        <button igxButton="icon" (click)="addProgress()">
+            <igx-icon fontSet="material">add</igx-icon>
+        </button>
+    </div>
+</div>
+```
 
-Following the simplest approach, we create a new theme that extends the [`igx-progress-circular-theme`]({environment:sassApiUrl}/index.html#function-igx-progress-circular-theme) and accepts the `$base-circle-color`, `$progress-circle-color` and `$text-color` parameters.
+Add the functions that add and remove value from the progress:
 
-```scss
-$custom-theme: igx-progress-circular-theme(
-    $base-circle-color: lightgray,
-    $progress-circle-color: rgb(32, 192, 17),
-    $text-color: lightgray
-);
+```typescript
+public currentValue: number;
+
+public ngOnInit() {
+    this.currentValue = 0;
+}
+
+public addProgress() {
+    this.currentValue += 10;
+    if (this.currentValue > 100) {
+        this.currentValue = 100;
+    }
+}
+
+public removeProgress() {
+    this.currentValue -= 10;
+    if (this.currentValue < 0) {
+        this.currentValue = 0;
+    }
+}
 ```
-The last step is to **include** the component mixins: 
+
+Add some styles to align items:
 
 ```scss
- @include igx-progress-circular($custom-theme);
+.custom-size {
+  width: 100px;
+  height: 100px;
+}
+
+.sample-content {
+  width: 300px;
+  display: flex;
+  align-items: center;
+  margin-top: 30px;
+}
 ```
 
->[!NOTE]
- >If the component is using an [`Emulated`](./themes/component-themes.md#view-encapsulation) ViewEncapsulation, it is necessary to `penetrate` this encapsulation using `::ng-deep`:
+#### Gradient Progress
 
- ```scss
-:host {
-  ::ng-deep {
-    @include igx-progress-circular($custom-theme);
-  }
-}
+You can use the [`IgxProgressBarGradientDirective`]({environment:angularApiUrl}/classes/igxcircularprogressbarcomponent.html#gradienttemplate) to colorize the progress bar with a gradient color.
```

Comment:
```suggestion
One way to customize the progress bar is to use a color gradient instead of a solid color.
This can be done in one of two ways - by using the [`IgxProgressBarGradientDirective`]({environment:angularApiUrl}/classes/igxcircularprogressbarcomponent.html#gradienttemplate) directive or by implementing a custom theme, albeit custom themes support up to two color stops.
```